src/pages/fresh_install/FreshInstallPage.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import '../fresh_install/FreshInstallPage.css';
+import path4 from "../../images/path4.svg";
+import classNames from 'classnames';
+
+export default class FreshInstallPage extends React.Component<{}, { selectedTemperatureScale: string }> {
+
+    handleButtonClick = (selectedTemperatureScale: 'C' | 'F') => {
+        if (this.state === null || !(this.state.selectedTemperatureScale === selectedTemperatureScale)) {
+            this.setState({
+                selectedTemperatureScale
+            });
+
+            localStorage.setItem('temperatureScale', selectedTemperatureScale);
```

Comment:
Following the SOLID principles, I'd recommend us to separate all persistence logic in a separated class from the UI logic that is present in this component.

Reply:
You are absolutely correct. I created a TemperatureScaleService inside services folder, so now every page that wants to get or set the value, could use their methods to do it.